.werft/platform-delete-preview-environments-cron.ts

```diff
@@ -37,48 +37,102 @@ async function deletePreviewEnvironments() {
     }
     werft.done("prep")
 
-    werft.phase("Fetching branches");
-    const branches = getAllBranches();
-    // During the transition from the old preview names to the new ones we have to check for the existence of both the old or new
-    // preview name patterns before it is safe to delete a namespace.
-    const expectedPreviewEnvironmentNamespaces = new Set(branches.flatMap(branch => [parseBranch(branch), expectedNamespaceFromBranch(branch)]));
-    werft.done("Fetching branches");
-
     werft.phase("Fetching previews");
     let previews: string[]
     try {
         previews = listAllPreviewNamespaces(CORE_DEV_KUBECONFIG_PATH, {});
-        previews.forEach(previewNs => werft.log("Fetching previews", previewNs))
+        previews.forEach(previewNs => werft.log("Fetching preview", previewNs));
+        werft.done("Fetching preview");
     } catch (err) {
-        werft.fail("Fetching previews", err)
+        werft.fail("Fetching preview", err)
     }
-    werft.done("Fetching previews");
+
+    werft.phase("Fetching outdated branches");
+    const branches = getAllBranches();
+    const outdatedPreviews = new Set(branches
+        .filter(branch => {
+          const lastCommit = exec(`git log ${branch} --since=$(date +%Y-%m-%d -d "5 days ago")`, { silent: true })
+          return lastCommit.length < 1
+        })
+        .map(branch => expectedNamespaceFromBranch(branch.replace('origin\/','').trim())))
+
+    const expectedPreviewEnvironmentNamespaces = new Set(branches.map(branch => expectedNamespaceFromBranch(branch)));
 
     werft.phase("deleting previews")
     try {
-        const previewsToDelete = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
-        // Trigger namespace deletion in parallel
-        const promises = previewsToDelete.map(preview => wipePreviewEnvironmentAndNamespace(helmInstallName, preview, CORE_DEV_KUBECONFIG_PATH, { slice: `Deleting preview ${preview}` }));
-        // But wait for all of them to finish before (or one of them to fail) before we continue
-        await Promise.all(promises)
+        const deleteDueToMissingBranch  = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
+        const deleteDueToNoCommitActivity  = previews.filter(ns => outdatedPreviews.has(ns))
+        const deleteDueToNoDBActivity      = previews.filter(ns => checkActivity(ns))
+        const previewsToDelete             = new Set([...deleteDueToMissingBranch, ...deleteDueToNoCommitActivity, ...deleteDueToNoDBActivity])
+
+        if (previewsToDelete.has("staging-main"))
+            previewsToDelete.delete("staging-main")
+
+        const dryRun = true
+        if (dryRun) {
+            previewsToDelete.forEach(preview => werft.log("would have deleted preview environment", preview))
+            werft.done("would have deleted preview environment")
```

Comment:
The slice id seems would here, I think it should have been `werft.done("deleting preview")`. Alternatively you can move the `done` till after the if/else

Reply:
Done